classes/class-swsales-metaboxes.php

```diff
@@ -409,26 +429,33 @@ public static function display_step_banner( $post ) {
 						<p class="description"><?php esc_html_e( 'Recommended: Leave checked so only users using your landing page will pay the sale price.', 'sitewide-sales' ); ?></p>
 					</td>
 				</tr>
-				<tr>
-					<th><label for="swsales_hide_banner_by_role"><?php esc_html_e( 'Hide Banner by Role', 'sitewide-sales' ); ?></label></th>
-					<td>
-						<input type="hidden" name="swsales_hide_banner_by_role_exists" value="1" />
-						<select multiple class="swsales_option" id="swsales_hide_banner_by_role_select" name="swsales_hide_banner_by_role[]">
+				<?php
+					// Old filter to hide banner by role. This is being deprecated in place of the field swsales_hide_for_roles.
+					$hide_for_roles = json_decode( $cur_sale->get_meta_value( 'swsales_hide_banner_by_role', '[]' ) );
+					if ( ! empty( $hide_for_roles ) ) { ?>
+						<tr>
+							<th><label for="swsales_hide_banner_by_role"><?php esc_html_e( 'Hide Banner by Role', 'sitewide-sales' ); ?></label></th>
```

Comment:
for needs to point to an id, so probably change to swsales_hide_banner_by_role_select

Reply:
A way to test is clicking on the label should select the associated input/field.